src/components/input/input.vue

```diff
@@ -0,0 +1,219 @@
+
+defaultVariantMixin(inputSize).verifyVariant(size.value)
+  ? ''
+  : (size.value = 'md');
+
+defaultVariantMixin(inputVariant).verifyVariant(props.variant)
+  ? ''
+  : (type.value = 'text');
+
+const inputClass= ref('input--' + type.value);
+
+// Watch for Password-show
+watch(
+  () => showPassword.value,
+  (newVal) => {
+    newVal
+      ? (variantMiddleware.value = 'text')
+      : (variantMiddleware.value = 'password');
+  }
+);
+
+
+function emitInput(e: EventTarget | null) {
+  if (e && 'value' in e) {
+    emit('update:modelValue', e.value)
+  }
+  return
+}
+
+function changePasswordVisibility() {
+  showPassword.value = !showPassword.value;
+}
+
+function toggleAccordian() {
+  accordianSwitch.value = !accordianSwitch.value;
+}
+
+function start(el: HTMLElement) {
+  el.style.height = el.scrollHeight + 'px';
+}
+function end(el: HTMLElement) {
+  el.style.height = '';
+}
+
+// dynamic component import
+const AsyncIcon = computed(() => {
+  {
+    const Icon = defineAsyncComponent(
+      () => import('@/components/icons/icons.vue')
+    );
+    return Icon;
+  }
+});
+
+//USE sentContent class for styling in parent
+
+</script>
+
+<template>
+  <div class="input-body">
+    <div :class="[size, 'input-body__top-wrapper', props.disabled ? 'disabled' : '']">
+      <label :for="labelRef" class="body1">
+        {{ props.label }}
+        <div v-if="accordianRef" class="icon-wrapper">
+          <AsyncIcon
+            v-if="accordianRef"
+           
+            class="dropdown"
+            :class="[accordianSwitch ? 'accordian__toggled' : '']"
+            icon="caret-down"
+            @click="toggleAccordian"
+          />
+        </div>
+      </label>
+
+      <transition
+        name="slot"
+        @enter="start"
+        @after-enter="end"
+        @before-leave="start"
+        @after-leave="end"
+      >
+        <slot
+          v-if="accordianSwitch"
+          name="sentContent"
+          class="sent-content"
+          appear
+        />
+      </transition>
+    </div>
+
+    <div
+      :disabled="props.disabled"
+      :class="[
+        size,
+        'input-wrapper',
+        inputClass,
+        props.isValid ? 'valid' : '',
+        props.isValid == false ? 'invalid' : '',
+        props.disabled ? 'disabled' : '',
+      ]"
+    >
+      <input
+        :id="labelRef"
+        autocomplete="off"
+        :type="variantMiddleware"
+        :placeholder="props.placeholder"
+        :disabled="props.disabled"
+        :value="props.modelValue"
+        @input="emitInput($event.target)"
+      >
+
+      <div
+        v-if="variantMiddleware == 'password' || props.variant == 'password'"
+        class="password-controls"
+      >
+        <AsyncIcon
+          v-if="showPassword == false"
+          icon="eye"
+          @click="changePasswordVisibility"
+        />
+        <AsyncIcon
+          v-if="showPassword == true"
+          icon="eye-slash"
+          @click="changePasswordVisibility"
```

Comment:
Would be nice with tabindex and `cursor: pointer` on the eye icon as well

Reply:
already added in new version

Reply:
currently thinking about autofocus to next icon becouse it loses focus when using eye